### PR TITLE
Ci setup

### PR DESCRIPTION
**Pull request by sameeul**
- Pytest harness
- GTest Harness
- Github workflow for Tests
- Github workflow for publishing to DockerHub and PyPi

### CODE REVIEW

File: .github/workflows/publish_docker.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+    
+    - name: Extract metadata (tags, labels) for Docker
+      id: meta
+      uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+      with:
+        images: sameeul/testrepo
```
Comment (hsidky) [inline .github/workflows/publish_docker.yml:46]: Should we keep `sameeul/testrepo` until we test then change it to the appropriate one, or?

File: .github/workflows/publish_pypi.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      run: docker build . -f .github/LinuxWheelBuilder  -t linuxbuilder
+
+    - name: Build ManyLinux Wheels 
+      run: docker run -v $(pwd):/io  linuxbuilder:latest
+    
+    - name: Install Dependencies
+      run: python -m pip install --upgrade twine requests
+
+    - name: Publish to PyPi
+      run: python -m twine upload dist/*
```
Comment (hsidky) [inline .github/workflows/publish_pypi.yml:29]: should we change this to test first?

Comment (hsidky) [reply .github/workflows/publish_pypi.yml:29]: Nevermind, I see the twine_repository below

**Review by hsidky — APPROVED** [review]
Approved. Please review my comments then I'll merge.

File: .github/workflows/publish_docker.yml (hunk 1 above)
Comment (sameeul) [reply .github/workflows/publish_docker.yml:46]: let's use my cred and test. And then update to PolusAI cred

**Pull request merged**